controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -1517,4 +1534,9 @@ protected int getTransactionEpoch(UUID txId) {
     abstract CompletableFuture<Data<T>> getWaitingRequestNode();
 
     abstract CompletableFuture<Void> deleteWaitingRequestNode();
+
+    abstract CompletableFuture<Void> createStartingSegmentNumberNode(int startingSegmentNumber);
```

Comment:
we can avoid these two methods if we store the starting segment number in segment index. 
That way we can avoid an additional network call into zookeeper.

Reply:
That's true, this method has been removed as now the starting segment number is stored in the first position of the `segmentIndex`.